docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -12,3 +12,116 @@ pageClass: migration-guide
+
+#### Before
+
+[`updateSettings()`](@/api/core.md#updatesettings) calls that included the [`data`](@/api/options.md#data) option automatically reset Handsontable's [index mappings](@/api/indexmapper.md) and [configuration options](@/guides/getting-started/setting-options.md) (`CellMeta`).
+
+#### After
+
+[`updateSettings()`](@/api/core.md#updatesettings) calls that include the [`data`](@/api/options.md#data) option don't automatically reset Handsontable's [index mappings](@/api/indexmapper.md) and [configuration options](@/guides/getting-started/setting-options.md) (`CellMeta`).
+
+This table summarizes the changes:
+
+| Type of call                                                                                                                                  | Before                                                                                                                                                                                                                                               | After                                                                                                                                                                                                                                                                                                                                                                |
+| --------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <span style="white-space: nowrap;">[`updateSettings()`](@/api/core.md#updatesettings)</span><br> with [`data`](@/api/options.md#data) defined | <ul><li>Always replaces [`data`](@/api/options.md#data) using [`loadData()`](@/api/core.md#loaddata)</li><li>[Options](@/guides/getting-started/setting-options.md) are reset</li><li>Index mappings are reset</li></ul>                             | <ul><li>At initialization, replaces [`data`](@/api/options.md#data) using [`loadData()`](@/api/core.md#loaddata)</li><li>After initialization, replaces [`data`](@/api/options.md#data) using [`updateData()`](@/api/core.md#updatedata)</li><li>[Options](@/guides/getting-started/setting-options.md) are not reset</li><li>Index mappings are not reset</li></ul> |
```

Comment:
I don't think that it is clear to the user, when we admit that one public API method (`updateSettings`) uses another public API method (`loadData()`, `updateData()`). For some reasons, we actually do that in code, but it is rare and the reader should not be expected to understand that.

I propose to replace:

```suggestion
| <span style="white-space: nowrap;">[`updateSettings()`](@/api/core.md#updatesettings)</span><br> with [`data`](@/api/options.md#data) defined | <ul><li>Replaces the data</li><li>Triggers the same hooks as [`loadData()`](@/api/core.md#loaddata)</li><li>[Options](@/guides/getting-started/setting-options.md) are reset</li><li>Index mappings are reset</li></ul>                             | <ul><li>Replaces the data</li><li>Triggers the same hooks as [`updateData()`](@/api/core.md#updatedata)</li><li>[Options](@/guides/getting-started/setting-options.md) are not reset</li><li>Index mappings are not reset</li></ul> |
```

As one of the changes in the above suggestion, I removed the sentence `At initialization, replaces [`data`](@/api/options.md#data) using [`loadData()`](@/api/core.md#loaddata)`, because I am not even sure what "at initialization" means. `updateSettings` is always called after the initialization. The `new Handsontable()` constructor performs initialization.

@jansiegel you might want to confirm here.

Reply:
@warpech The point is, that `updateSettings` uses `loadData` at initialization, as the constructor calls it directly (the data is initially set up using `loadData`, it cannot use `updateData` as there's no data to be updated). After that, it's always `updateData`.

Reply:
Ok, but this is our implementation detail that `new Handsontable()` constructor uses a public method `updateSettings` internally. For a consumer of our API, who is the audience of this migration guide, `updateSettings` is always used after initialization.

Reply:
> Ok, but this is our implementation detail that `new Handsontable()` constructor uses a public method `updateSettings` internally. For a consumer of our API, who is the audience of this migration guide, `updateSettings` is always used after initialization.

Partially that's true, but on the other hand, the hooks from the `updateSettings` method still get triggered publicly at initialization and users may want to know why they were different the first time.

Reply:
From the user's perspective the initialization hooks are called from `new Handsontable()`, not from `updateSettings`. The user is not expected to know or care if `new Handsontable()` uses `updateSettings`.

If there was any change in the initialization hooks, it would deserve its own entry in the migration guide. But there is no difference in the initialization hooks, so it's not worth mentioning. What am I missing?

Reply:
> But where they different?

True, this doesn't change, it was added to paint a bigger picture, but maybe the migration guide is not the right place to do that.

Reply:
@warpech @jansiegel – so we're going for:

<img width="845" alt="5" src="https://user-images.githubusercontent.com/29247866/157736349-fd3702fa-a47f-45f3-89fa-3618148f478d.png">

Reply:
Nearly. In my proposal (first message in the thread), I proposed to write:

| Before                                                                                                                                                                                                                                               | After                                                                                                                                                                                                                                                                                                                                                                |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| <ul><li>Replaces the data</li><li>Triggers the same hooks as [`loadData()`](@/api/core.md#loaddata)</li><li>...</li></ul>                             | <ul><li>Replaces the data</li><li>Triggers the same hooks as [`updateData()`](@/api/core.md#updatedata)</li><li>...</li></ul> |

Because it should not interest the user of the API that `updateSettings` actually uses `loadData`/`updateData`. That is an implementation detail. What matters is the external effects.

Reply:
@warpech - done: [8cdf83e](https://github.com/handsontable/handsontable/pull/9271/commits/8cdf83eeb6ffc7e66e49d09395dc0e6fc62a0a5f)